examples/js/controls/EditorControls.js

```diff
@@ -18,35 +19,209 @@ THREE.EditorControls = function ( object, domElement ) {
 
 	var scope = this;
 	var vector = new THREE.Vector3();
+	var matrix = new THREE.Matrix3();
 
 	var STATE = { NONE: -1, ROTATE: 0, ZOOM: 1, PAN: 2 };
 	var state = STATE.NONE;
 
 	var center = this.center;
-	var normalMatrix = new THREE.Matrix3();
-	var pointer = new THREE.Vector2();
-	var pointerOld = new THREE.Vector2();
+
+	// pointer data
+
+	var touches = [];
+
+	// pointers are expressed in -1 to 1 coordinate space relative to domElement.
+
+	var pointers = [ new THREE.Vector2(), new THREE.Vector2() ];
+	var pointersOld = [ new THREE.Vector2(), new THREE.Vector2() ];
+	var pointersDelta = [ new THREE.Vector2(), new THREE.Vector2() ];
 
 	// events
 
 	var changeEvent = { type: 'change' };
 
+	// hepler functions
+
+	var getClosestPoint = function( point, pointArray ) {
+
+		if ( pointArray[ 0 ].distanceTo( point) < pointArray[ 1 ].distanceTo( point) ) {
+
+			return pointArray[ 0 ];
+		
+		}
+
+		return pointArray[ 1 ];
+
+	};
+
+	var setPointers = function( event ) {
+
+		// Set pointes from mouse/touch events and convert to -1 to 1 coordinate space.
+
+		var parentRect = event.path[ 0 ].getBoundingClientRect();
+
+		// Filter touches that originate from the same element as the event.   
```

Comment:
This is necessary to support multiple viewports and have ability to use touches to manipulate multiple viewports simultaneously. Without it, both instances of EditorControls would receive combined touches from all viewports.